src/snowflake/snowpark/session.py

```diff
@@ -1330,6 +1330,7 @@ def create_dataframe(
         self,
         data: Union[List, Tuple, "pandas.DataFrame"],
         schema: Optional[Union[StructType, List[str]]] = None,
+        chunk_size: Optional[int] = 100000,
```

Comment:
I don't like adding parameter of a helper function to the caller's signature. Should we just not expose this parameter?  The only concern is `chunk_size` is not the size of the file to be uploaded, but the number of rows in a chunk, so not exposing this parameter means the issue reported could still occur with this "fix".